sync/syncer.go

```diff
@@ -215,12 +215,13 @@ func (s *Syncer) Close() {
 	close(s.exit)
 	close(s.forceSync)
 	s.peers.Close()
+	s.syncLock.Lock()
+	s.syncLock.Unlock()
```

Comment:
why not close inside the lock? this could be still be a bit racy

Reply:
the lock is taken to make sure close has been processed and the loop has exited. after this happens there can be not more requests to sync